flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/converter/PostgresRowConverter.java

```diff
@@ -91,7 +91,10 @@ private JdbcDeserializationConverter createPostgresArrayConverter(ArrayType arra
                 final Object[] array = (Object[]) Array.newInstance(elementClass, in.length);
                 for (int i = 0; i < in.length; i++) {
                     array[i] =
-                            elementConverter.deserialize(((PGobject) in[i]).getValue().getBytes());
+                            elementConverter.deserialize(
+                                    in[i] instanceof byte[]
+                                            ? in[i]
+                                            : ((PGobject) in[i]).getValue().getBytes());
```

Comment:
It would be good to add a comment on why the conditional logic is now needed.

Reply:
@snuyanzin WDYT?

Reply:
Thanks for the thorough explanation and good catch. Since it looks like a new functionality, I would expect to also have a dedicated test case for `byte[]`. WDYT?

Reply:
I see. Adding a small comment explaining the pg situation would help others read the line again.

Reply:
done